www/md_base/guanlecoja/config.coffee

```diff
@@ -88,4 +88,33 @@ gulp.task 'icons', ->
         ))
         .pipe(gulp.dest(path.join(config.dir.build, 'icons')))
 
+gulp.task 'apiproxy', ->
+    # this is a task for developing, it proxy api request to http://nine.buildbot.net
+    argv = require('minimist')(process.argv)
+    argv.apiserver ?= 'nine.buildbot.net'
+    argv.apiport ?= 8021
+    argv.server ?= 'http://localhost:8020'
+
+    fs = require 'fs'
+    path = require 'path'
+    http = require 'http'
+    httpProxy = require 'http-proxy'
+    proxy = httpProxy.createProxyServer({})
+    proxy.on 'proxyReq', (proxyReq, req, res, options) ->
+        delete proxyReq.removeHeader('Origin')
+        delete proxyReq.removeHeader('Referer')
+    proxy.on 'proxyRes', (proxyRes, req, res) ->
+        proxyRes.headers['Access-Control-Allow-Origin'] = '*'
+        console.log "[Proxy] #{req.method} #{req.url}"
+
+    server = http.createServer (req, res) ->
+        if req.url.match /^\/api/
+            proxy.web req, res, {target: 'http://' + argv.apiserver}
+        else if req.url.match /^\/ws/
+            proxy.ws req, res, {target: 'ws://' + argv.apiserver}
+        else
+            proxy.web req, res, {target: argv.server}
```

Comment:
You perferctly rephrased what I meant.

The only thing is that we should require this only for the dev mode. The proxy server shall not be even launched for default build, and prod build.
we could also add a proxy task that is not a dependancy of dev.

gulp dev proxy --url http://nine.buildbot.net/

This will address your concerns I think.